repository/src/main/java/org/alfresco/repo/domain/schema/script/DeleteNotExistsV3Executor.java

```diff
@@ -0,0 +1,501 @@
+        {
+            sqlBuilder.append(" LIMIT " + batchSize);
+        }
+        else
+        {
+            sqlBuilder.append(" OFFSET 0 ROWS FETCH FIRST " + batchSize + " ROWS ONLY");
+        }
+
+        return sqlBuilder.toString();
+    }
+
+    @Override
+    protected Long processPrimaryTableResultSet(PreparedStatement primaryPrepStmt, PreparedStatement[] secondaryPrepStmts,
+            PreparedStatement deletePrepStmt, Set<Long> deleteIds, String primaryTableName, String primaryColumnName,
+            Pair<String, String>[] tableColumn) throws SQLException
+    {
+        Long primaryId = null;
+        Long minSecId = 0L;
+        Long maxSecId = 0L;
+
+        // Set all rows retrieved from the primary table as our potential ids to delete
+        Set<Long> potentialIdsToDelete = new HashSet<Long>();
+        Long minPotentialId = 0L;
+        Long maxPotentialId = 0L;
+        try (ResultSet resultSet = primaryPrepStmt.getResultSet())
+        {
+            while (resultSet.next())
+            {
+                primaryId = resultSet.getLong(primaryColumnName);
+                potentialIdsToDelete.add(primaryId);
+
+                minPotentialId = (minPotentialId == 0L || primaryId < minPotentialId) ? primaryId : minPotentialId;
+                maxPotentialId = primaryId > maxPotentialId ? primaryId : maxPotentialId;
+            }
+        }
+
+        if (potentialIdsToDelete.size() == 0)
+        {
+            // Nothing more to do
+            return primaryId;
+        }
+
+        int rowsInBatch = potentialIdsToDelete.size();
+        processedCounter = processedCounter + rowsInBatch;
+
+        // Get a combined list of the ids present in the secondary tables
+        SecondaryResultsInfo secondaryResultsInfo = getSecondaryResults(secondaryPrepStmts, tableColumn, minPotentialId,
+                maxPotentialId);
+
+        Set<Long> secondaryResults = secondaryResultsInfo.getValues();
+
+        if (secondaryResultsInfo.getSize() > 0)
+        {
+            minSecId = secondaryResultsInfo.getMinValue();
+            maxSecId = secondaryResultsInfo.getMaxValue();
+
+            // From our potentialIdsToDelete list, remove all non-eligible ids: any id that is in a secondary table or
+            // any ID past the last ID we were able to access in the secondary tables (maxSecId)
+            Iterator<Long> it = potentialIdsToDelete.iterator();
+            while (it.hasNext())
+            {
+                Long id = it.next();
+                if (id > maxSecId || secondaryResults.contains(id))
+                {
+                    it.remove();
+                }
+            }
+
+            // The next starting primary ID for the next batch will either be the next last id evaluated from the
+            // primary table or, in case the secondary queries did not get that far, the last secondary table id
+            // evaluated (maxSecId)
+            primaryId = primaryId < maxSecId ? primaryId : maxSecId;
```

Comment:
This is `min(primaryId, maxSecId)` and above we had something equivalent to `max(primaryId, maxPotentialId)`.  I think using these tertiary operators is fine, but this was just something that went through my mind as I read the code.